### PR TITLE
main: Autodetect project if one is not specified

### PR DESCRIPTION
**Pull request by gvso**
This uses the compute metadata package to detect the ID of the project on which the service is deployed. If the tool is not being run on GCE, it tries to detect the project ID by executing a `gcloud` command to get the default project

### CODE REVIEW

File: cmd/operator/main.go [hunk 1]
```diff
@@ -237,3 +250,28 @@ func printHealthCriteria(logger *logrus.Logger, healthCriteria []config.HealthCr
...
+	if metadata.OnGCE() {
+		v, err := metadata.ProjectID()
+		if err != nil {
+			return "", errors.Wrapf(err, "error when getting project ID from compute metadata: %v", err)
+		}
+		return v, nil
+	}
+
+	// Try to get project ID by retrieving default value in gcloud.
+	cmd := exec.Command("gcloud", "config", "get-value", "core/project")
```
Comment (ahmetb) [inline cmd/operator/main.go:264]: add `-q` to make it not show potential prompts

File: cmd/operator/main.go [hunk 2]
```diff
@@ -237,3 +250,28 @@ func printHealthCriteria(logger *logrus.Logger, healthCriteria []config.HealthCr
...
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return "", errors.Wrap(err, "error when running gcloud command to get default project")
```
Comment (ahmetb) [inline cmd/operator/main.go:269]: would be good to return the error message as well. so capture stderr in another buffer

File: cmd/operator/main.go [hunk 3]
```diff
@@ -237,3 +250,28 @@ func printHealthCriteria(logger *logrus.Logger, healthCriteria []config.HealthCr
...
+	}
+
+	v := strings.Trim(out.String(), "\n")
```
Comment (ahmetb) [inline cmd/operator/main.go:272]: strings.TrimSpace

File: cmd/operator/main.go [hunk 4]
```diff
@@ -251,25 +251,31 @@ func printHealthCriteria(logger *logrus.Logger, healthCriteria []config.HealthCr
...
 
 	// Try to get project ID by retrieving default value in gcloud.
-	cmd := exec.Command("gcloud", "config", "get-value", "core/project")
-	var out bytes.Buffer
-	cmd.Stdout = &out
+	cmd := exec.Command("gcloud", "config", "get-value", "core/project", "-q")
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
```
Comment (ahmetb) [inline cmd/operator/main.go:268]: not necessary but you can do this in the future:

var stdout, stderr bytes.Buffer
cmd.Stdout, cmd.Stderr = &stdout, &stderr

Comment (gvso) [reply cmd/operator/main.go]: I made this change too

**Pull request merged**